configuration/postgres/postgres.go

```diff
@@ -0,0 +1,432 @@
+	}
+	return p.subscribeToChannel(ctx, pgNotifyChannels, req, handler)
+}
+
+func (p *ConfigurationStore) Unsubscribe(ctx context.Context, req *configuration.UnsubscribeRequest) error {
+	p.configLock.Lock()
+	defer p.configLock.Unlock()
+	for _, v := range p.ActiveSubscriptions {
+		if v.uuid == req.ID {
+			break
+		} else {
+			return fmt.Errorf("unable to find subscription with ID : %v", req.ID)
+		}
+	}
+	if oldStopChan, ok := p.subscribeStopChanMap[req.ID]; ok {
+		delete(p.subscribeStopChanMap, req.ID)
+		close(oldStopChan.(chan struct{}))
+		for k, v := range p.ActiveSubscriptions {
+			if v.uuid == req.ID {
+				pgChannel := fmt.Sprintf(unlistenTemplate, k)
+				conn, err := p.client.Acquire(ctx)
+				if err != nil {
+					p.logger.Errorf("error acquiring connection:", err)
+					return fmt.Errorf("error acquiring connection: %s ", err)
+				}
+				defer conn.Release()
+				_, err = conn.Exec(ctx, pgChannel)
+				if err != nil {
+					p.logger.Errorf("error listening to channel:", err)
+					return fmt.Errorf("error listening to channel: %s", err)
+				}
+				delete(p.ActiveSubscriptions, k)
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("unable to find subscription with ID : %v", req.ID)
+}
+
+func (p *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, command string, channel string, subscription string, stop chan struct{}) {
+	conn, err := p.client.Acquire(ctx)
+	if err != nil {
+		p.logger.Errorf("error acquiring connection:", err)
+		return
+	}
+	defer conn.Release()
+	if _, err = conn.Exec(ctx, command); err != nil {
+		p.logger.Errorf("error listening to channel:", err)
+		return
+	}
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			if !pgconn.Timeout(err) && !errors.Is(ctx.Err(), context.Canceled) {
+				p.logger.Errorf("error waiting for notification:", err)
+			}
+			return
+		}
+		p.handleSubscribedChange(ctx, handler, notification, channel, subscription)
+	}
+}
+
+func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler configuration.UpdateHandler, msg *pgconn.Notification, channel string, subscriptionID string) {
+	defer func() {
+		if err := recover(); err != nil {
+			p.logger.Errorf("panic in handlesubscribedchange method and recovered: %s", err)
+		}
+	}()
+	payload := make(map[string]interface{})
+	err := json.Unmarshal([]byte(msg.Payload), &payload)
+	if err != nil {
+		p.logger.Errorf("error in unmarshal: ", err)
+		return
+	}
+	var key, value, version string
+	m := make(map[string]string)
+	// trigger should encapsulate the row in "data" field in the notification
+	row := reflect.ValueOf(payload[payloadDataKey])
+	if row.Kind() == reflect.Map {
+		for _, k := range row.MapKeys() {
+			v := row.MapIndex(k)
+			strKey := k.Interface().(string)
+			switch strings.ToLower(strKey) {
+			case "key":
+				key = v.Interface().(string)
```

Comment:
Can this cast fail at all? Should we check if cast is OK first?
Same for other casts below...

Reply:
As discussed offline, panic will handle any edge case exceptions raised due to failing string case